homeassistant/components/pushbullet/sensor.py

```diff
@@ -213,3 +222,7 @@ def run(self):
             self.listener.run_forever()
         finally:
             self.listener.close()
+
+    def shutdown(self, event: Event | None = None) -> None:
+        """Shut down the thread."""
+        self.join()
```

Comment:
This only joins the thread, it doesn't shut it down.

https://docs.python.org/3/library/threading.html#threading.Thread.join

Reply:
After a little digging I realized that the `Listener` class inherits from Thread so I don't need to create a thread separately. I rewrote the `PushBulletNotificationProvider` class and now it should start the thread inside `Listener` when HASS starts and it will close on config entry unload.
https://github.com/richard-better/pushbullet.py/blob/3aed153c5f5356d47e86212851df652d6c907e62/pushbullet/listener.py#L15